Build fails if GDAL is disabled

**Description of the problem**

If GDAL is disable or not linked, the build fails with following errors: 
```
/mnt/home/tiandong/softwares/GMT/gmt-dev/src/gmt_plot.c: In function ‘gmt_importproj4’:
/mnt/home/tiandong/softwares/GMT/gmt-dev/src/gmt_plot.c:5799:3: error: unknown type name ‘OGRSpatialReferenceH’
   OGRSpatialReferenceH  hSRS;
   ^~~~~~~~~~~~~~~~~~~~
/mnt/home/tiandong/softwares/GMT/gmt-dev/src/gmt_plot.c:5800:3: error: unknown type name ‘OGRErr’
   OGRErr eErr = OGRERR_NONE;
   ^~~~~~
/mnt/home/tiandong/softwares/GMT/gmt-dev/src/gmt_plot.c:5800:17: error: ‘OGRERR_NONE’ undeclared (first use in this function)
   OGRErr eErr = OGRERR_NONE;
                 ^~~~~~~~~~~
/mnt/home/tiandong/softwares/GMT/gmt-dev/src/gmt_plot.c:5800:17: note: each undeclared identifier is reported only once for each function it appears in
/mnt/home/tiandong/softwares/GMT/gmt-dev/src/gmt_plot.c:5813:10: warning: implicit declaration of function ‘OSRNewSpatialReference’ [-Wimplicit-function-declaration]
   hSRS = OSRNewSpatialReference(NULL);
          ^~~~~~~~~~~~~~~~~~~~~~
/mnt/home/tiandong/softwares/GMT/gmt-dev/src/gmt_plot.c:5814:15: warning: implicit declaration of function ‘OSRImportFromEPSG’ [-Wimplicit-function-declaration]
   if ((eErr = OSRImportFromEPSG(hSRS, EPSGID)) != OGRERR_NONE) {
               ^~~~~~~~~~~~~~~~~
/mnt/home/tiandong/softwares/GMT/gmt-dev/src/gmt_plot.c:5818:15: warning: implicit declaration of function ‘OSRExportToProj4’ [-Wimplicit-function-declaration]
   if ((eErr = OSRExportToProj4(hSRS, &pszResult)) != OGRERR_NONE) {
               ^~~~~~~~~~~~~~~~
/mnt/home/tiandong/softwares/GMT/gmt-dev/src/gmt_plot.c:5823:3: warning: implicit declaration of function ‘CPLFree’ [-Wimplicit-function-declaration]
   CPLFree(pszResult);
   ^~~~~~~
/mnt/home/tiandong/softwares/GMT/gmt-dev/src/gmt_plot.c:5824:3: warning: implicit declaration of function ‘OSRDestroySpatialReference’ [-Wimplicit-function-declaration]
   OSRDestroySpatialReference(hSRS);
   ^~~~~~~~~~~~~~~~~~~~~~~~~~
```

Like #703, we need to check if `HAVE_GDAL` is defined.

--- a/src/gmt_plot.c
+++ b/src/gmt_plot.c
@@ -5774,6 +5774,7 @@ GMT_LOCAL void wipe_substr(char *str1, char *str2) {
 }
 #endif
 
+#ifdef HAVE_GDAL
 char *gmt_importproj4 (struct GMT_CTRL *GMT, char *pStr, int *scale_pos) {
 	/* Take a PROJ.4 projection string or EPSG code and try to find the equivalent -J syntax
 	   scale_pos is position on the return string where starts the scale sub-string.
@@ -6244,6 +6245,7 @@ char *gmt_importproj4 (struct GMT_CTRL *GMT, char *pStr, int *scale_pos) {
 	pStrOut = strdup(opt_J);
 	return pStrOut;
 }
+#endif
 
 char *gmt_export2proj4 (struct GMT_CTRL *GMT) {
 	char *pStrOut = NULL;

--- a/src/gmt_prototypes.h
+++ b/src/gmt_prototypes.h
@@ -190,7 +190,9 @@ EXTERN_MSC struct GMT_GRID *gmt_duplicate_grid (struct GMT_CTRL *GMT, struct GMT
 
 EXTERN_MSC int gmt_ps_append (struct GMT_CTRL *GMT, char *source, unsigned int mode, FILE *dest);
 EXTERN_MSC char *gmt_export2proj4 (struct GMT_CTRL *GMT);
+#ifdef HAVE_GDAL
 EXTERN_MSC char *gmt_importproj4 (struct GMT_CTRL *GMT, char *szProj4, int *scale_pos);
+#endif
 EXTERN_MSC int gmt_strip_layer (struct GMTAPI_CTRL *API, int nlayers);
 EXTERN_MSC void gmt_textpath_init (struct GMT_CTRL *GMT, struct GMT_PEN *BP, double Brgb[]);
 EXTERN_MSC void gmt_draw_map_rose (struct GMT_CTRL *GMT, struct GMT_MAP_ROSE *mr);

